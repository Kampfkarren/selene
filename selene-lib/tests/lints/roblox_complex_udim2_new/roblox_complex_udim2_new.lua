UDim2.new(0)
UDim2.new(0.5)
UDim2.new(1)
UDim2.new(2)
UDim2.new(a)
UDim2.new(1, 1)
UDim2.new(1, 2)
UDim2.new(a, b)
UDim2.new(1, a)
UDim2.new(1, 1, 1)
UDim2.new(a, b, c)
UDim2.new(1, 1, 1, 1)
UDim2.new(a, b, c, d)
UDim2.fromOffset(1, 1)
UDim2.fromScale(1, 1)
UDim2.new()
UDim2.new(1, 0, 1, 0)
UDim2.new(0, 0, 1, 0)
UDim2.new(0, 5, 1, 5)
UDim2.new(0, 5, 0, 1)
UDim2.new(0, 0, 0, 1)
UDim2.new(0, 1, 0, 1)
UDim2.new(0, a, 0, 1)
