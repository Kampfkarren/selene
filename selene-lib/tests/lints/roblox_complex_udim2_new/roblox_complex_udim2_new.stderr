error[roblox_complex_udim2_new]: this UDim2.new call only sets scale, and can be simplified using UDim2.fromScale
   ┌─ roblox_complex_udim2_new.lua:17:1
   │
17 │ UDim2.new(1, 0, 1, 0)
   │ ^^^^^^^^^^^^^^^^^^^^^
   │
   = try: UDim2.fromScale(1, 1)

error[roblox_complex_udim2_new]: this UDim2.new call only sets scale, and can be simplified using UDim2.fromScale
   ┌─ roblox_complex_udim2_new.lua:18:1
   │
18 │ UDim2.new(0, 0, 1, 0)
   │ ^^^^^^^^^^^^^^^^^^^^^
   │
   = try: UDim2.fromScale(0, 1)

error[roblox_complex_udim2_new]: this UDim2.new call only sets offset, and can be simplified using UDim2.fromOffset
   ┌─ roblox_complex_udim2_new.lua:20:1
   │
20 │ UDim2.new(0, 5, 0, 1)
   │ ^^^^^^^^^^^^^^^^^^^^^
   │
   = try: UDim2.fromOffset(5, 1)

error[roblox_complex_udim2_new]: this UDim2.new call only sets offset, and can be simplified using UDim2.fromOffset
   ┌─ roblox_complex_udim2_new.lua:21:1
   │
21 │ UDim2.new(0, 0, 0, 1)
   │ ^^^^^^^^^^^^^^^^^^^^^
   │
   = try: UDim2.fromOffset(0, 1)

error[roblox_complex_udim2_new]: this UDim2.new call only sets offset, and can be simplified using UDim2.fromOffset
   ┌─ roblox_complex_udim2_new.lua:22:1
   │
22 │ UDim2.new(0, 1, 0, 1)
   │ ^^^^^^^^^^^^^^^^^^^^^
   │
   = try: UDim2.fromOffset(1, 1)

